### ISSUE TITLE
False positive for `Style/FileOpen` when passing IO object to external API

### ISSUE THREAD

**Issue by ydakuka** [rubocop/rubocop#14958]
## Steps to reproduce the problem

```ruby
def json_key_io
  File.open(File.absolute_path(json_key_file_path))
end

Google::Auth::ServiceAccountCredentials.make_creds(
  json_key_io: json_key_io,
  scope: Google::Apis::SheetsV4::AUTH_SPREADSHEETS,
)
```

Running RuboCop reports:

```
Style/FileOpen: File.open without a block may leak a file descriptor; use the block form.
```

## Expected behavior

`Style/FileOpen` should not register an offense when `File.open` is used to return or pass an IO object to another API that requires an open IO instance.

In such cases, using block syntax is not possible because the file would be closed before the receiving library can consume it.

## RuboCop version

```
$ rubocop -V
1.85.0 (using Parser 3.3.10.1, Prism 1.9.0, rubocop-ast 1.49.0, analyzing as Ruby 3.4, running on ruby 3.4.8) [x86_64-linux]
  - rubocop-capybara 2.22.1
  - rubocop-factory_bot 2.28.0
  - rubocop-migration 0.7.1
  - rubocop-performance 1.26.1
  - rubocop-rails 2.34.3
  - rubocop-rake 0.7.1
  - rubocop-rspec 3.9.0
  - rubocop-rspec_rails 2.32.0
  - rubocop-thread_safety 0.7.3
```

**Comment by sferik** [comment]
Sorry for the trouble! I introduced this cop in https://github.com/rubocop/rubocop/pull/14942.

@koic Please let me know if I can help with the fix.
